inventory_management_system_api/repositories/catalogue_category.py

```diff
@@ -201,25 +203,32 @@ def list(self, parent_id: Optional[str], session: ClientSession = None) -> List[
         return [CatalogueCategoryOut(**catalogue_category) for catalogue_category in catalogue_categories]
 
     def _is_duplicate_catalogue_category(
-        self, parent_id: Optional[str], code: str, session: ClientSession = None
+        self,
+        parent_id: Optional[str],
+        code: str,
+        catalogue_category_id: CustomObjectId = None,
+        session: ClientSession = None,
     ) -> bool:
         """
         Check if a catalogue category with the same code already exists within the parent category.
 
         :param parent_id: The ID of the parent catalogue category which can also be `None`.
         :param code: The code of the catalogue category to check for duplicates.
+        :param catalogue_category_id: The ID of the catalogue category to check if the duplicate
+         catalogue category found is itself.
         :param session: PyMongo ClientSession to use for database operations
         :return: `True` if a duplicate catalogue category code is found, `False` otherwise.
         """
         logger.info("Checking if catalogue category with code '%s' already exists within the parent category", code)
         if parent_id:
             parent_id = CustomObjectId(parent_id)
 
-        return (
-            self._catalogue_categories_collection.find_one({"parent_id": parent_id, "code": code}, session=session)
-            is not None
+        catalogue_category_with_code = self._catalogue_categories_collection.find_one(
```

Comment:
Calling this `catalogue_category` should be just fine I think
```suggestion
        catalogue_categorye = self._catalogue_categories_collection.find_one(
```